service/src/main_impl.rs

```diff
@@ -755,159 +761,21 @@ fn spawn_worker_for_shard_polling<InitializationHandler>(
 	});
 }
 
-fn print_events(events: Vec<Event>) {
+fn print_events<R, H>(events: Vec<EventRecord<R, H>>, parentchain_id: ParentchainId)
+where
+	R: Debug,
+{
```

Comment:
we can now print any event per parentchain, even unknown ones. however, we currently depend on a solochain runtime Event type which most likely fails to decode here when testing against a parachain parentchain

guess we should put the only Runtime dependencies into ita-stf-parentchain where customization can take place

Reply:
yeah, it works against solochain, as seen in CI logs. but not against parachain

Reply:
Yes, I think we should put the runtime deps into app libs. Introduce a trait `PrintEvents`, implement it per runtime events in the app libs, and have some generic code in main, with the `Events: PrintEvents` trait bound.

Reply:
#1506